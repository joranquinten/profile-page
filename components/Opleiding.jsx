import React from 'react';
import styled from 'styled-components';

import { Card, Table } from 'antd';

import { colors } from '../styled/themes/default';

import { education } from '../data/nl';

const columns = [
  {
    title: 'Wanneer',
    dataIndex: 'date',
    key: 'date',
  },
  {
    title: 'Opleiding of certificaat',
    dataIndex: 'description',
    key: 'description',
  },
];

const StyledOpleiding = styled.div`
  .ant-table {
    border-color: ${colors.blackDarkened};
    color: ${colors.white};

  .ant-table-tbody > tr > td {
    border-color: ${colors.blackDarkened};
  }

  .ant-table-tbody > tr:hover:not(.ant-table-expanded-row) > td {
    color: ${colors.primaryColor};
    background-color: ${colors.blackDarkened};
  }

  .ant-card-head-title {
    color: ${colors.tertiaryColor};
  }
`;

const Opleiding = () => (
  <StyledOpleiding>
    <Card title={education.title}>
      <Table
        className="ant-table-override"
        dataSource={education.items}
        columns={columns}
        bordered={false}
        pagination={false}
        showHeader={false}
        size="small"
      />
    </Card>
  </StyledOpleiding>
);

export default Opleiding;
