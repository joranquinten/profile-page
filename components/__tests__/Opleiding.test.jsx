import { shallow } from 'enzyme';

import Opleiding from '../Opleiding';

jest.mock('../../data/nl.js');

describe('<Opleiding />', () => {
  it('should render without crashing', () => {
    expect(shallow(<Opleiding />)).toMatchSnapshot();
  });
});
