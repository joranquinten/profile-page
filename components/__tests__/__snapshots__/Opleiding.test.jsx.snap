// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Opleiding /> should render without crashing 1`] = `
<Opleiding__StyledOpleiding>
  <Card
    title="educationTitle"
  >
    <Table
      bordered={false}
      className="ant-table-override"
      columns={
        Array [
          Object {
            "dataIndex": "date",
            "key": "date",
            "title": "Wanneer",
          },
          Object {
            "dataIndex": "description",
            "key": "description",
            "title": "Opleiding of certificaat",
          },
        ]
      }
      dataSource={
        Array [
          Object {
            "date": "January 2018",
            "description": "educationFirstDescription",
            "key": "1",
          },
          Object {
            "date": "February 2018",
            "description": "educationSecondDescription",
            "key": "2",
          },
        ]
      }
      indentSize={20}
      loading={false}
      locale={Object {}}
      pagination={false}
      rowKey="key"
      showHeader={false}
      size="small"
      sortDirections={
        Array [
          "ascend",
          "descend",
        ]
      }
      useFixedHeader={false}
    />
  </Card>
</Opleiding__StyledOpleiding>
`;
